### ISSUE TITLE
`nvim` process sometimes not quitting with Neovide

### ISSUE THREAD

**Issue by chrisgrieser** [neovide/neovide#1595]
**Describe the bug**
I noticed that sometimes, the `nvim` process created by neovide is not quitting, even if I quit the Neovim app. Since it seems that on start Neovide creates a new `nvim` process, after a while I end up with multiple `nvim` processes, all taking up RAM, even if Neovide isn't running anymore:
<img width="917" alt="Pasted image 2022-10-12 15 06 54" src="https://user-images.githubusercontent.com/73286100/195351531-dd7018b2-4865-4bf2-b227-174a0ada3d47.png">

**To Reproduce**
I am not sure how exactly to reproduce this. sometimes it happens, sometimes not.

**Expected behavior**
When the last instance of neovide quits, neovim-processes should quit as well.

**Desktop (please complete the following information):**
```
macOS 12.5.1 (M1)
neovim 0.8.0 (homebrew)
Neovide 0.10.1 (homebrew)
```

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
[neovide_rCURRENT.log](https://github.com/neovide/neovide/files/9765165/neovide_rCURRENT.log)

**Additional context**
I mostly start neovide via the CLI, not the `Neovide.app`.

**Comment by WillEhrendreich** [comment]
yep, I've got it too, makes it annoying to do config iteration if the old process is still hanging around afterwards. i was hoping to find a solution or submit an issue. you beat me too it. :)

**Comment by chrisgrieser** [comment]
Just wanna mention that this is really getting out of hand. with lazy.nvim, it's recommended not to re-source files but restart nvim as a whole. with neovide, this results in *a lot* of leftover processes...

![CleanShot 2023-04-10 at 12 56 31@2x](https://user-images.githubusercontent.com/73286100/230889136-bd084b25-84ad-4962-8f70-d2b6973793d0.png)

**Comment by chrisgrieser** [comment]
This is really getting out of hand. with lazy.nvim, it's recommended to restart instead of re-sourcing, so you end up with this really quickly.

![CleanShot 2023-04-10 at 12 56 31@2x](https://user-images.githubusercontent.com/73286100/230889226-b0562a2e-ee21-449f-9339-6c9d56d058d5.png)

**Comment by fredizzimo** [comment]
This should be fixed by https://github.com/neovide/neovide/pull/2463, and included in the Neovide 0.13.0 release.
